.travis.yml

```diff
@@ -12,31 +12,37 @@ matrix:
       env:
         - DESC="acceptance tests"
         - CMD="mvn install --projects acceptance-tests --also-make --activate-profiles all --batch-mode --show-version --errors"
+        - SONAR_COMMAND = ""
 
     - jdk: oraclejdk8
       env:
         - DESC="findbugs"
         - CMD="mvn install findbugs:check --projects '!scala-unit-tests,!jmh-scala-tests,!jmh-tests' --activate-profiles all -DskipTests=true --batch-mode --show-version --errors"
+        - SONAR_COMMAND = ""
 
     - jdk: oraclejdk8
       env:
         - DESC="checkstyle"
         - CMD="mvn install checkstyle:check --activate-profiles all -DskipTests=true --batch-mode --show-version --errors"
+        - SONAR_COMMAND = ""
 
     - jdk: oraclejdk8
       env:
         - DESC="unit tests"
         - CMD="mvn install --batch-mode --show-version --errors"
+        - SONAR_COMMAND = "sonar:sonar -Dsonar.host.url=https://sonarqube.com -Dsonar.login=${SONARQUBE_TOKEN} -Dsonar.organization=eclipse-collections"
 
     - jdk: oraclejdk8
       env:
         - DESC="compile jmh-tests and performance-tests"
         - CMD="mvn install --projects jmh-tests,performance-tests --also-make --activate-profiles all -DskipTests=true --batch-mode --show-version --errors"
+        - SONAR_COMMAND = ""
 
 script:
-  if [[ "$TRAVIS_EVENT_TYPE" == "cron" && "$DESC" == "unit tests" ]];
-  then mvn install --batch-mode --show-version --errors sonar:sonar -Dsonar.host.url=https://sonarqube.com -Dsonar.login=${SONARQUBE_TOKEN} -Dsonar.organization=eclipse-collections ;
-  else eval $CMD;
+  if ["$TRAVIS_EVENT_TYPE" == "cron"]; then
+      eval $CMD $SONAR_COMMAND;
```

Comment:
@motlin @nikhilnanivadekar FWIW another option something like:
```
matrix:
  fast_finish: true
  include:
    ...
    - jdk: oraclejdk8
      env:
        - DESC="unit tests"
        - CMD="mvn install --batch-mode --show-version --errors"
    ...

script:
  - SONAR_COMMAND=""
  - if ["$TRAVIS_EVENT_TYPE" == "cron"]; then SONAR_COMMAND="..."; fi
  - eval $CMD;
```

And coming back to suggestion about `switch` - something like:
```
matrix:
  fast_finish: true
  include:
    ...
    - jdk: oraclejdk8
      env:
        - DESC="unit tests"
    ...

script:
  - if ["$TRAVIS_EVENT_TYPE" == "cron"]; then SONAR_COMMAND="..."; fi
  - case "$DESC" in
  "unit tests")
    CMD="..."
    ;;
  esac
  - eval $CMD $SONAR_COMMAND
```
maybe the last one might require to move the whole script into file to avoid issues with `.travis.ym` such as one command per `-` and so on:
```
script: travis.sh
```

Also please recall that this PR is not about beautification of `.travis.yml`, but mainly about fix of problem that is stated in description.

Hope this helps.